### PR TITLE
Create navigation.mdx

### PR DESCRIPTION
**Pull request by staton-hyse11**
Added navigation pattern documentation.

We need a content review to make sure this makes sense before merging

### CODE REVIEW

File: docs/06-patterns/navigation.mdx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+**When to Use:**  
+The megamenu provides access to levels 1, 2, and 3 of the Grafana information architecture. Levels beyond this should be handled by tabs on a page (level 4) or inline elements using headings, steppers or other design elements (level 5 and up).
+
+**Usage Guidelines:**  
+- Except in rare cases, cloud applications should not be placed at level 1 of the megamenu. Instead, they should be grouped together with similar applications.
+- Ensure all megamenu entries have a corresponding page or place to go for consistent breadcrumb behavior.
+- Level 1 content must have an icon next to it for visual consistency; other levels do not utilize icons.
+- Utilize a landing page for all overview/category pages.
+- Ensure all megamenu entries are in sentence case.
```
Comment (eledobleefe) [inline docs/06-patterns/navigation.mdx:23]: FMPOV, I'd avoid repeating the rules. For example, I see this point as a "Dos": 'Do ensure all megamenu entries are in sentence case'. I'd leave the "Usage Guidelines" for general indications.

File: docs/06-patterns/navigation.mdx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+## Breadcrumbs Component Documentation
+
+Grafana’s topnav breadcrumbs are a key part of the overall navigation experience. They are the very top most popular part of the 2022-2023 navigation redesign, and so it is critical to the user experience that they behave in a consistent and uniform manner.
+
+**When to Use:**  
+Breadcrumbs are automatically generated by the page hierarchy, so they are used on all pages.
+
+**Usage Guidelines:**  
+- Breadcrumbs should be used for information architecture hierarchy and not for history.
+- Utilize a separate page for anything reflected as a breadcrumb.
```
Comment (eledobleefe) [inline docs/06-patterns/navigation.mdx:52]: As I said previously, I'd avoid repeating the rules. Could it be a good idea that "Usage Guidelines" comprises "Dos" and "Don'ts" sections? They are all focused on usage guidance 🤔

File: docs/06-patterns/navigation.mdx [hunk 3]
```diff
@@ -0,0 +1,65 @@
+---
+title: Navigation
+description: Navigation patterns are crucial for ensuring users can seamlessly move through an application.
+hide_title: true
+---
+
+# Navigation <Badge text='ready' color='green'></Badge>
+Navigation patterns are crucial for ensuring users can seamlessly move through an application. In Grafana, two key components play a vital role in navigation: the Megamenu and Breadcrumbs. Let's delve into how these components contribute to effective navigation patterns.
```
Comment (eledobleefe) [inline docs/06-patterns/navigation.mdx:8]: Maybe just me, but I missed the article here 🤷 
```suggestion
Navigation patterns are crucial for ensuring users can seamlessly move through an application. In Grafana, two key components play a vital role in navigation: the Megamenu and the Breadcrumbs. Let's delve into how these components contribute to effective navigation patterns.
```

**Review by JoaoSilvaGrafana — COMMENTED** [review]
Agree with Laura's comments on duplicate content. Also would it be worth having an image/schematic to explain what the Megamenu is? And I guess breadcrumbs as well

**Review by amy-super — APPROVED** [review]
I'm cool with feedback shared by @eledobleefe and @JoaoSilvaGrafana on the duplication (even though I do err on the side of being overly clear). 

Regarding the images, I think that's something we can add at some point if it's a lot of work. I'm more interested in getting _something_ out there for people to use as guidelines over perfecting the presentation of it.

File: docs/06-patterns/navigation.mdx (hunk 1 above)
Comment (staton-hyse11) [reply docs/06-patterns/navigation.mdx]: Addressed this 👍

File: docs/06-patterns/navigation.mdx (hunk 2 above)
Comment (staton-hyse11) [reply docs/06-patterns/navigation.mdx]: Addressed this 👍

File: docs/06-patterns/navigation.mdx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+**When to Use:**  
+The megamenu provides access to levels 1, 2, and 3 of the Grafana information architecture. Levels beyond this should be handled by tabs on a page (level 4) or inline elements using headings, steppers or other design elements (level 5 and up).
+
+### Usage Guidelines  
+To ensure users have a consistent and expected experience when they are navigating through Grafana, it is important to follow the following Dos and Don'ts.
+
+**Dos:**  
+- Utilize a landing page for all overview/category pages.
+- Ensure all entries have a corresponding page or place to go for consistent breadcrumb behavior.
+- Be sure to incluee an icon next to any level 1 content
```
Comment (eledobleefe) [inline docs/06-patterns/navigation.mdx:24]:
```suggestion
- Be sure to include an icon next to any level 1 content. 
```

File: docs/06-patterns/navigation.mdx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+## Breadcrumbs Component Documentation
+
+Grafana’s topnav breadcrumbs are a key part of the overall navigation experience. They are the very top most popular part of the 2022-2023 navigation redesign, and so it is critical to the user experience that they behave in a consistent and uniform manner.
+
+**When to Use:**  
+Breadcrumbs are automatically generated by the page hierarchy, so they are used on all pages.
+
+### Usage Guidelines:  
+Breadcrumbs provide users with a critical indicator of where they are within the product space. Having breadcrumbs behave consistently throughout the product, ensures that users have good experience and are able to understand where they are regardless of what part of the product they are currently viewing. The following guidelines will ensure we are provide a consisten and expected experience.
```
Comment (eledobleefe) [inline docs/06-patterns/navigation.mdx:49]:
```suggestion
Breadcrumbs provide users with a critical indicator of where they are within the product space. Having breadcrumbs behave consistently throughout the product, ensures that users have a good experience and are able to understand where they are regardless of what part of the product they are currently viewing. The following guidelines will ensure we are providing a consistent and expected experience.
```

**Review by eledobleefe — APPROVED** [review]
Just some nits. Otherwise, LGTM 👌

**Pull request merged**